### ISSUE TITLE
invalid IRR object flagged as notfound

### ISSUE THREAD

**Issue by troy2914** [irrdnet/irrd#461]
in 4.1.1
RPKI object in RIPE  - RPKI enabled
object in JPIRR - RPKI enabled object is invalid
object in RADB - RPKI disabled

result is  it shows all 3 objects.   Expected it would not show the JPIRR invalid object.
Note this JPIRR object may be cleaned up soon.
whois -h irrd01-beta.ip.gin.ntt.net 192.47.117.0/24
```
route:          192.47.117.0/24
descr:          RPKI ROA for 192.47.117.0/24 / AS2518
remarks:        This AS2518 route object represents routing data retrieved
                from the RPKI. This route object is the result of an automated
                RPKI-to-IRR conversion process performed by IRRd.
max-length:     24
origin:         AS2518
source:         RPKI  # Trust Anchor: apnic
route:          192.47.117.0/24
descr:          MIDORI NET 1
origin:         AS45674
notify:         noc@mesh.ad.jp
mnt-by:         MAINT-AS2518
changed:        n-horiuchi@za.biglobe.co.jp 20200601
source:         JPIRR
rpki-ov-state:  not_found # No ROAs found, or RPKI validation not enabled for source
route:          192.47.117.0/24
descr:          MIDORI NET 1
origin:         AS45674
notify:         noc@mesh.ad.jp
mnt-by:         MESH
changed:        t-mabuchi@biglobe.co.jp 20150717
source:         RADB
rpki-ov-state:  not_found # No ROAs found, or RPKI validation not enabled for source
```

The database on this host also did not flag that prefix as invalid from any sources.

**Comment by troy2914** [comment]
hmm,  this does not seem stable....   Looking at the db i see:  
```
irrd=> select rpsl_pk,rpki_status,source from  rpsl_objects where rpsl_pk like '192.47.117.0/24AS%' limit 4;
         rpsl_pk           | rpki_status | source
----------------------------+-------------+--------
 192.47.117.0/24AS2518/ML24 | valid       | RPKI
 192.47.117.0/24AS45674     | invalid     | JPIRR
 192.47.117.0/24AS45674     | invalid     | RADB
```
do whois,  only shows the 1 valid RPKI route (despite RADB being set for 
```
           rpki_excluded: true
```

did a 
```
psql -d irrd -t -A -q -c "select rpsl_pk,source from rpsl_objects where rpki_status = 'invalid'::rpkistatus;"
```
that prefix does not show up.
a few minutes later
```
irrd=> select rpsl_pk,rpki_status,source from  rpsl_objects where rpsl_pk like '192.47.117.0/24AS%' limit 4;
          rpsl_pk           | rpki_status | source
----------------------------+-------------+--------
 192.47.117.0/24AS45674     | not_found   | RADB
 192.47.117.0/24AS2518/ML24 | valid       | RPKI
 192.47.117.0/24AS45674     | not_found   | JPIRR
``` 
and the whois shows all 3 objects again.

**Comment by troy2914** [comment]
starting running this every 2 minutes:
```
psql -d irrd -t -A -q -c "select rpsl_pk,rpki_status,source from  rpsl_objects where rpsl_pk in ('192.47.117.0/24AS45674','192.47.117.0/24AS2518/ML24');" > debug/status.`date +%Y%m%d-%H%M%S`
```
times in utc,  115 is broken, 111 is good, here are when the states changed (was broken originally):
-rw-r--r-- 1 irrd irrd 111 Jan 29 16:36 status.20210129-163601
-rw-r--r-- 1 irrd irrd 115 Jan 29 16:34 status.20210129-163401
and
-rw-r--r-- 1 irrd irrd 115 Jan 29 17:32 status.20210129-173201
-rw-r--r-- 1 irrd irrd 111 Jan 29 17:30 status.20210129-173001

**Comment by troy2914** [comment]
and flipped back to good around 18:35
-rw-r--r-- 1 irrd irrd 111 Jan 29 18:36 status.20210129-183601
-rw-r--r-- 1 irrd irrd 115 Jan 29 18:34 status.20210129-183401

I am sensing a theme here...

**Comment by troy2914** [comment]
-rw-r--r-- 1 irrd irrd 115 Jan 29 19:32 status.20210129-193201
-rw-r--r-- 1 irrd irrd 111 Jan 29 19:30 status.20210129-193001

**Comment by troy2914** [comment]
btw, seems to be about a 6362 prefix difference between the good and bad states (# of invalid's reported).

**Comment by mxsasha** [comment]
This is rather puzzling, but does look like a bug. The hourly cycle suggests that at least one part of this is related to the periodic RPKI importer. But I'm not really sure what would make it flip back shortly after.

Could you share the logs? Particularly what happened between known good and bad times and vice versa (with a margin of a few minutes before and after). That should tell us what processes are running around that time.

**Comment by troy2914** [comment]
the 2 minutes is the frequency of my check.  So the most recent flip was 2021-02-01 12:36 and 12:38, here are the irrd server.log covering those 2 and 2 minutes before 12:36:
```
2021-02-01 12:35:44,134 irrd[120378]: [irrd.server.http.server#INFO] HTTP request served: ::ffff:89.248.168.108: code 501, message Unsupported method ('HEAD')
2021-02-01 12:35:44,173 irrd[120378]: [irrd.server.http.server#INFO] HTTP request served: ::ffff:89.248.168.108: "HEAD / HTTP/1.0" 501 -
2021-02-01 12:36:12,011 irrd[1044]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for TC from serial 59976 on whois.bgp.net.br:43
2021-02-01 12:36:26,960 irrd[1062]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for INTERNAL from serial 31416 on rr.ntt.net:43
2021-02-01 12:36:26,960 irrd[1063]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for LACNIC from serial 10815 on irr.lacnic.net:43
2021-02-01 12:36:41,991 irrd[1082]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for RIPE-NONAUTH from serial 50131969 on 193.0.6.135:4444
2021-02-01 12:36:41,991 irrd[1083]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for ARIN from serial 70618 on rr.arin.net:4444
2021-02-01 12:36:57,007 irrd[1104]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for RADB from serial 5108491 on whois.radb.net:43
2021-02-01 12:37:08,332 irrd[130499]: [irrd.mirroring.mirror_runners_import#INFO] RPKI status updated for all routes, 5120 newly valid, 110 newly invalid, 71086 newly not_found routes, 0 emails sent to contacts of newly invalid authoritative objects
2021-02-01 12:37:12,031 irrd[1123]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for NTTCOM from serial 1245109 on rr.ntt.net:43
2021-02-01 12:37:12,033 irrd[1124]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for RIPE from serial 50131970 on 193.0.6.135:4444
2021-02-01 12:37:12,034 irrd[1125]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for ALTDB from serial 74226 on whois.altdb.net:43
2021-02-01 12:37:12,384 irrd[1124]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation RIPE/50131973/ADD/as-set/AS-ALTUS
2021-02-01 12:37:12,977 irrd[1124]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation RIPE/50131977/ADD/route/89.111.184.0/21AS5537, RPKI status INVALID
2021-02-01 12:37:13,272 irrd[1124]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation RIPE/50131983/ADD/route/89.111.177.0/24AS5537, RPKI status INVALID
2021-02-01 12:37:13,280 irrd[1124]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation RIPE/50131985/ADD/route/89.111.176.0/21AS5537, RPKI status INVALID
2021-02-01 12:37:13,287 irrd[1124]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation RIPE/50131987/ADD/route/89.111.176.0/20AS5537, RPKI status INVALID
2021-02-01 12:37:13,289 irrd[1124]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation RIPE/50132002/ADD/mntner/ES-HECANET-1-MNT
2021-02-01 12:37:27,059 irrd[1147]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for LEVEL3 from serial 816261 on rr.Level3.net:43
2021-02-01 12:37:27,060 irrd[1146]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for BELL from serial 249037 on 205.207.237.48:43
2021-02-01 12:37:27,065 irrd[1148]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for RGNET from serial 467 on whois.rg.net:43
2021-02-01 12:37:42,083 irrd[1273]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for JPIRR from serial 189748 on 202.12.30.249:43
2021-02-01 12:37:42,083 irrd[1272]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for APNIC from serial 8393798 on nrtm.apnic.net:43003
2021-02-01 12:37:42,098 irrd[1274]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for ARIN-NONAUTH from serial 70232 on rr.arin.net:4444
2021-02-01 12:37:42,489 irrd[1272]: [irrd.mirroring.mirror_runners_import#ERROR] An error occurred while attempting a mirror update or initial import for APNIC: [Errno 104] Connection reset by peer
2021-02-01 12:37:53,031 irrd[120379]: [irrd.storage.preload#INFO] Completed updating preload store from thread <PreloadUpdater(Thread-7754, started 139865895016192)>
2021-02-01 12:37:57,129 irrd[1299]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for BBOI from serial 18623 on 66.216.21.35:43
2021-02-01 12:37:57,129 irrd[1300]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for AFRINIC from serial 1032059 on 196.216.2.20:43003
2021-02-01 12:38:15,740 irrd[120379]: [irrd.storage.preload#INFO] Completed updating preload store from thread <PreloadUpdater(Thread-7755, started 139865778210560)>
```

-rw-r--r-- 1 irrd irrd 115 2021-02-01 12:38:01.505208099 +0000 status.20210201-123801
-rw-r--r-- 1 irrd irrd 111 2021-02-01 12:36:01.437475501 +0000 status.20210201-123601

from the hour earlier:
```
2021-02-01 11:41:07,206 irrd[125677]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for TC from serial 59976 on whois.bgp.net.br:43
2021-02-01 11:41:22,222 irrd[125694]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for INTERNAL from serial 31416 on rr.ntt.net:43
2021-02-01 11:41:22,227 irrd[125695]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for LACNIC from serial 10815 on irr.lacnic.net:43
2021-02-01 11:41:37,246 irrd[125715]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for RIPE-NONAUTH from serial 50131655 on 193.0.6.135:4444
2021-02-01 11:41:37,246 irrd[125716]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for ARIN from serial 70617 on rr.arin.net:4444
2021-02-01 11:41:52,275 irrd[125736]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for RADB from serial 5108449 on whois.radb.net:43
2021-02-01 11:41:52,375 irrd[125736]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation RADB/5108449/ADD/route/67.231.184.0/24AS7413, RPKI status NOT_FOUND
2021-02-01 11:41:52,376 irrd[125736]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation RADB/5108450/ADD/route/208.205.128.0/22AS7413, RPKI status NOT_FOUND
2021-02-01 11:41:52,376 irrd[125736]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation RADB/5108451/ADD/route/63.97.220.0/23AS7413, RPKI status NOT_FOUND
2021-02-01 11:41:52,377 irrd[125736]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation RADB/5108452/ADD/route/63.97.219.0/24AS7413, RPKI status NOT_FOUND
2021-02-01 11:41:52,377 irrd[125736]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation RADB/5108453/ADD/route/65.196.229.0/24AS7413, RPKI status NOT_FOUND
2021-02-01 11:41:52,378 irrd[125736]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation RADB/5108454/ADD/route/216.75.252.0/24AS7413, RPKI status NOT_FOUND
2021-02-01 11:41:52,378 irrd[125736]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation RADB/5108455/ADD/route/216.75.250.0/24AS7413, RPKI status NOT_FOUND
2021-02-01 11:41:52,379 irrd[125736]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation RADB/5108456/ADD/route/65.196.234.0/24AS7413, RPKI status NOT_FOUND
2021-02-01 11:41:52,380 irrd[125736]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation RADB/5108457/ADD/route/216.75.248.0/24AS7413, RPKI status NOT_FOUND
2021-02-01 11:41:52,380 irrd[125736]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation RADB/5108458/ADD/route/65.219.166.0/24AS7413, RPKI status NOT_FOUND
2021-02-01 11:41:52,381 irrd[125736]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation RADB/5108459/ADD/route/216.75.245.0/24AS7413, RPKI status NOT_FOUND
2021-02-01 11:41:52,381 irrd[125736]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation RADB/5108460/ADD/route/208.254.174.0/24AS7413, RPKI status NOT_FOUND
2021-02-01 11:42:06,426 irrd[104285]: [irrd.mirroring.mirror_runners_import#INFO] RPKI status updated for all routes, 58992 newly valid, 3229 newly invalid, 6477 newly not_found routes, 0 emails sent to contacts of newly invalid authoritative objects
2021-02-01 11:42:06,440 irrd[125221]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation RIPE/50131646/DEL/as-set/AS-OGONE
2021-02-01 11:42:06,440 irrd[125221]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation RIPE/50131648/ADD/aut-num/AS51682
2021-02-01 11:42:07,290 irrd[125777]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for NTTCOM from serial 1245109 on rr.ntt.net:43
2021-02-01 11:42:07,295 irrd[125778]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for RIPE from serial 50131657 on 193.0.6.135:4444
2021-02-01 11:42:07,296 irrd[125779]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for ALTDB from serial 74226 on whois.altdb.net:43
2021-02-01 11:42:07,637 irrd[125778]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation RIPE/50131657/ADD/aut-num/AS8775
2021-02-01 11:42:07,638 irrd[125778]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation RIPE/50131659/ADD/aut-num/AS47957
2021-02-01 11:42:07,664 irrd[125778]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation RIPE/50131678/ADD/aut-num/AS8775
2021-02-01 11:42:22,318 irrd[125800]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for LEVEL3 from serial 816251 on rr.Level3.net:43
2021-02-01 11:42:22,318 irrd[125801]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for RGNET from serial 467 on whois.rg.net:43
2021-02-01 11:42:22,320 irrd[125799]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for BELL from serial 249037 on 205.207.237.48:43
2021-02-01 11:42:37,343 irrd[125928]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for JPIRR from serial 189748 on 202.12.30.249:43
2021-02-01 11:42:37,344 irrd[125929]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for ARIN-NONAUTH from serial 70232 on rr.arin.net:4444
2021-02-01 11:42:37,351 irrd[125927]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for APNIC from serial 8393798 on nrtm.apnic.net:43003
2021-02-01 11:42:37,750 irrd[125927]: [irrd.mirroring.mirror_runners_import#ERROR] An error occurred while attempting a mirror update or initial import for APNIC: [Errno 104] Connection reset by peer
2021-02-01 11:42:52,371 irrd[125952]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for BBOI from serial 18623 on 66.216.21.35:43
2021-02-01 11:42:52,371 irrd[125953]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for AFRINIC from serial 1032035 on 196.216.2.20:43003
2021-02-01 11:42:52,551 irrd[120379]: [irrd.storage.preload#INFO] Completed updating preload store from thread <PreloadUpdater(Thread-7735, started 139865895016192)>
2021-02-01 11:43:21,844 irrd[120379]: [irrd.storage.preload#INFO] Completed updating preload store from thread <PreloadUpdater(Thread-7736, started 139865778210560)>
2021-02-01 11:46:07,689 irrd[126206]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for TC from serial 59976 on whois.bgp.net.br:43
2021-02-01 11:46:22,643 irrd[126223]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for INTERNAL from serial 31416 on rr.ntt.net:43
2021-02-01 11:46:22,643 irrd[126224]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for LACNIC from serial 10815 on irr.lacnic.net:43
2021-02-01 11:46:37,671 irrd[126245]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for ARIN from serial 70617 on rr.arin.net:4444
2021-02-01 11:46:37,678 irrd[126244]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for RIPE-NONAUTH from serial 50131679 on 193.0.6.135:4444
2021-02-01 11:46:52,696 irrd[126268]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for RADB from serial 5108461 on whois.radb.net:43
2021-02-01 11:47:07,719 irrd[126287]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for NTTCOM from serial 1245109 on rr.ntt.net:43
2021-02-01 11:47:07,719 irrd[126289]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for ALTDB from serial 74226 on whois.altdb.net:43
2021-02-01 11:47:07,721 irrd[126288]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for RIPE from serial 50131681 on 193.0.6.135:4444
2021-02-01 11:47:09,203 irrd[126288]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation RIPE/50131699/ADD/route/85.208.153.0/24AS48789, RPKI status VALID
2021-02-01 11:47:09,334 irrd[126288]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation RIPE/50131703/ADD/route/185.2.144.0/23AS205005, RPKI status INVALID
2021-02-01 11:47:09,334 irrd[126288]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation RIPE/50131704/ADD/mntner/SWHL-MNT
2021-02-01 11:47:22,738 irrd[126309]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for BELL from serial 249037 on 205.207.237.48:43
2021-02-01 11:47:22,741 irrd[126311]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for RGNET from serial 467 on whois.rg.net:43
2021-02-01 11:47:22,743 irrd[126310]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for LEVEL3 from serial 816251 on rr.Level3.net:43
2021-02-01 11:47:37,762 irrd[126436]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for JPIRR from serial 189748 on 202.12.30.249:43
2021-02-01 11:47:37,762 irrd[126435]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for APNIC from serial 8393798 on nrtm.apnic.net:43003
2021-02-01 11:47:37,767 irrd[126437]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for ARIN-NONAUTH from serial 70232 on rr.arin.net:4444
2021-02-01 11:47:38,166 irrd[126435]: [irrd.mirroring.mirror_runners_import#ERROR] An error occurred while attempting a mirror update or initial import for APNIC: [Errno 104] Connection reset by peer
2021-02-01 11:47:43,552 irrd[120379]: [irrd.storage.preload#INFO] Completed updating preload store from thread <PreloadUpdater(Thread-7737, started 139865778210560)>
2021-02-01 11:47:52,794 irrd[126463]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for AFRINIC from serial 1032035 on 196.216.2.20:43003
2021-02-01 11:47:52,800 irrd[126461]: [irrd.mirroring.mirror_runners_import#INFO] Retrieving NRTM updates for BBOI from serial 18623 on 66.216.21.35:43
2021-02-01 11:47:53,440 irrd[126463]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation AFRINIC/1032035/ADD/route/41.180.136.0/22AS36916, RPKI status VALID
2021-02-01 11:47:53,766 irrd[126463]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation AFRINIC/1032036/ADD/route/41.180.144.0/22AS36916, RPKI status VALID
2021-02-01 11:47:54,004 irrd[126463]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation AFRINIC/1032037/ADD/route/41.180.152.0/22AS36916, RPKI status VALID
2021-02-01 11:47:54,247 irrd[126463]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation AFRINIC/1032038/ADD/route/41.180.164.0/22AS36916, RPKI status VALID
2021-02-01 11:47:54,480 irrd[126463]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation AFRINIC/1032039/ADD/route/41.180.168.0/22AS36916, RPKI status VALID
2021-02-01 11:47:54,713 irrd[126463]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation AFRINIC/1032040/ADD/route/41.180.172.0/22AS36916, RPKI status VALID
2021-02-01 11:47:54,946 irrd[126463]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation AFRINIC/1032041/ADD/route/41.180.184.0/22AS36916, RPKI status VALID
2021-02-01 11:47:55,176 irrd[126463]: [irrd.mirroring.nrtm_operation#INFO] Completed NRTM operation AFRINIC/1032042/ADD/route/41.180.192.0/22AS36916, RPKI status VALID
2021-02-01 11:48:28,682 irrd[120379]: [irrd.storage.preload#INFO] Completed updating preload store from thread <PreloadUpdater(Thread-7738, started 139865778210560)>
```

where the queries was run at
-rw-r--r-- 1 irrd irrd 111 2021-02-01 11:44:01.768609985 +0000 status.20210201-114401
-rw-r--r-- 1 irrd irrd 115 2021-02-01 11:42:01.636888756 +0000 status.20210201-114201

**Comment by troy2914** [comment]
and question about what I am seeing in roa_objects,  all the rows show the same created date.  A recent date at that.   
```
2021-02-01 15:26:41.560961+00
```
this was during the good state hour.  Interesting that datestamp does not like up with the switch from bad to good.

```
-rw-r--r-- 1 irrd irrd 111 2021-02-01 15:42:01.573774590 +0000 status.20210201-154201
-rw-r--r-- 1 irrd irrd 115 2021-02-01 15:40:01.506027766 +0000 status.20210201-154001
```

ah, but at that time in the log there was a:
```
2021-02-01 15:26:41,559 irrd[27115]: [irrd.mirroring.mirror_runners_import#INFO] Running full ROA import from: https://rpki.gin.ntt.net/api/export.json, SLURM file:///etc/rpki-slurm.json
```

**Comment by troy2914** [comment]
more RPKI oddness:
```
 210.130.209.0/24AS32787    | not_found   | JPIRR
 210.130.209.0/24AS32787    | not_found   | RADB
```
yet, roa_object shows a 
```
 68363af2-8d57-4d28-9a72-3706e7fab149 | 210.130.0.0/16 | 2497 |         16 | apnic        |          4 | 2021-02-01 15:26:41.560961+00
```

and whois shows:
```
route:          210.130.0.0/16
descr:          IIJ IPv4 BLOCK ( AS2497 )
origin:         AS2497
mnt-by:         MAINT-AS2497
changed:        w-ippei@iij.ad.jp 20141010
source:         RADB
rpki-ov-state:  not_found # No ROAs found, or RPKI validation not enabled for source

route:          210.130.0.0/16
descr:          RPKI ROA for 210.130.0.0/16 / AS2497
remarks:        This AS2497 route object represents routing data retrieved
                from the RPKI. This route object is the result of an automated
                RPKI-to-IRR conversion process performed by IRRd.
max-length:     16
origin:         AS2497
source:         RPKI  # Trust Anchor: apnic

route:          210.130.209.0/24
descr:          Prolexic Customer Route
origin:         AS32787
mnt-by:         MAINT-AS2497
changed:        shuhei-o@iij.ad.jp 20200716
source:         JPIRR
rpki-ov-state:  not_found # No ROAs found, or RPKI validation not enabled for source

route:          210.130.209.0/24
descr:          Prolexic Customer Route
origin:         AS32787
mnt-by:         MAINT-AS2497
changed:        noc@iij.ad.jp 20151124
source:         RADB
rpki-ov-state:  not_found # No ROAs found, or RPKI validation not enabled for source

```

now RADB (along with NTTCOM and an internal db) has rpki_excluded enabled,  but I would have expected the JPIRR object to be suppressed.

**Comment by mxsasha** [comment]
I'm still digging into this, but it seems to be an issue in the rpki_excluded setting - on my test instance I enabled it on RADB, and right after that the RADB and JPIRR object became not_found.

**Comment by troy2914** [comment]
is your test instance also show the flipping between not_found and invalid on a roughly hour basis?

**Comment by mxsasha** [comment]
Yes, and I found the bug: https://github.com/irrdnet/irrd/blob/master/irrd/storage/database_handler.py#L260-L271

If a route object with identical route and origin exists in multiple sources, and is invalid, and some but not all sources that have the identical object are RPKI excluded, these objects will flip between not_found and invalid on every RPKI import run (every hour by default). This might be specific to cases where RPKI exclusion settings are changed after the object is already in the DB.

The reason is that the query I linked updates based on RPSL primary key, which is only unique per source. So in the case of a route object existing in JPIRR and RADB, while RADB is excluded, one RPKI run will detect the RADB invalid object, decide it should be not_found because of exclusion, and incorrectly update both objects to not_found. The next run detects the JPIRR not_found object, decides that one should be invalid, and incorrectly updates both objects to invalid. And this way it flips back and forth every hour.

Probably the right fix is to use the actual primary key of the rows, which is entirely unique, and still allows the RPKI status updates to happen in bulk. Will work on that in the next few days and make a new release after I confirm it works.

Oh, and on this question:
> and question about what I am seeing in roa_objects, all the rows show the same created date. A recent date at that.

This is correct. On RPKI import, the ROA table is emptied and reloaded. This is fairly efficient because it happens in bulk, reliable, and saves the importer from having to run a full comparison between the current and new state. It's done in a single transaction, so other processes will never actually ever see an empty ROA table.
